app/javascript/spotlight/admin/crop.js

```diff
@@ -2,18 +2,26 @@ import { addImageSelector } from 'spotlight/admin/add_image_selector'
 import Core from 'spotlight/core'
 
 export default class Crop {
-  constructor(cropArea) {
+  constructor(cropArea, iiifFields = null, preserveAspectRatio = true) {
     this.cropArea = cropArea;
     this.cropArea.data('iiifCropper', this);
     this.cropSelector = '[data-cropper="' + cropArea.data('cropperKey') + '"]';
     this.cropTool = $(this.cropSelector);
-    this.formPrefix = this.cropTool.data('form-prefix');
-    this.iiifUrlField = $('#' + this.formPrefix + '_iiif_tilesource');
-    this.iiifRegionField = $('#' + this.formPrefix + '_iiif_region');
-    this.iiifManifestField = $('#' + this.formPrefix + '_iiif_manifest_url');
-    this.iiifCanvasField = $('#' + this.formPrefix + '_iiif_canvas_id');
-    this.iiifImageField = $('#' + this.formPrefix + '_iiif_image_id');
-
+    this.preserveAspectRatio = preserveAspectRatio;
+    if(iiifFields == null) {
+      this.formPrefix = this.cropTool.data('form-prefix');
+      this.iiifUrlField = $('#' + this.formPrefix + '_iiif_tilesource');
+      this.iiifRegionField = $('#' + this.formPrefix + '_iiif_region');
+      this.iiifManifestField = $('#' + this.formPrefix + '_iiif_manifest_url');
+      this.iiifCanvasField = $('#' + this.formPrefix + '_iiif_canvas_id');
+      this.iiifImageField = $('#' + this.formPrefix + '_iiif_image_id');
+    } else {
+      this.iiifUrlField = iiifFields["iiifUrlField"];
+      this.iiifRegionField = iiifFields["iiifRegionField"];
+      this.iiifManifestField = iiifFields["iiifManifestField"];
+      this.iiifCanvasField = iiifFields["iiifCanvasField"];
+      this.iiifImageField = iiifFields["iiifImageField"];
+    }
```

Comment:
Did you explore extending crop/otherwise breaking this out into a separate class? It seems like we may be able to drop this and one other conditional if we did. Maybe we'd have to create a cropper factory? I'm not sure if that's worth it or not.

Reply:
Oh, I see, I misread this the first time sorry. You've added `iiifFields` here to accept the data from `getIIIFObject` that gets the main page widget data and transfers it to the cropper in the modal via the constructor.

Could we rewrite the data for `dataCropperDiv` in `CroppableModal#attachModalLoadBehavior` rather than passing this around as a separate argument? We might need another piece of data and to abstract away what the `'#' + this.formPrefix` is doing now. That or maybe a lightweight extension to this class is the way to go. I just think the only mandatory parameter in the constructor being mostly ignored if the second argument is present should likely be untangled.

Reply:
I tried rewriting the constructor specifically.  Most of the code no longer relies on formPrefix but there is the upload field that does depend on it.  The upload id part is only used on the exhibit/masthead pages and not in the cropping modal. I could also work on changing that to the new method so thoughts welcome.

Reply:
I think this is great, thanks.